src/app/DIAmeterApplication.cpp

```diff
@@ -73,241 +80,871 @@ int DIAmeterApplication::main(const vector<string>& input_files, const string in
+		 active_peptide_queue->setElutionWindow(0);
+		 active_peptide_queue->setPeptideCentric(false);
+		 active_peptide_queue->SetBinSize(bin_width_, bin_offset_);
+		 active_peptide_queue->SetOutputs(NULL, &locations, Params::GetInt("top-match"), true, output_file, NULL, highest_ms2_mz);
+
+		 // initialize fields required for output
+		 // Not sure if it is necessary, will check later
+		 const vector<SpectrumCollection::SpecCharge>* spec_charges = spectra->SpecCharges();
+		 int* sc_index = new int(-1);
+		 FLOAT_T sc_total = (FLOAT_T)spec_charges->size();
+		 int print_interval = Params::GetInt("print-search-progress");
+		 // Keep track of observed peaks that get filtered out in various ways.
+		 long int num_range_skipped = 0;
+		 long int num_precursors_skipped = 0;
+		 long int num_isotopes_skipped = 0;
+		 long int num_retained = 0;
+
+		 // This is the main search loop.
+		 // DIAmeter supports spectrum centric match report only
+		 if (Params::GetBool("peptide-centric-search")) { carp(CARP_FATAL, "Spectrum-centric match only!"); }
+
+		 ObservedPeakSet observed(bin_width_, bin_offset_, Params::GetBool("use-neutral-loss-peaks"), Params::GetBool("use-flanking-peaks") );
+
+		 // Note:
+		 // We don't traverse the collection of SpecCharge, which is sorted by neutral mass and if the neutral mass is equal, sort by the MS2 scan.
+		 // Notice that in the DIA setting, each different neutral mass correspond to a (scan-win, charge) pair.
+		 // Therefore, we divide the collection of SpecCharge into different chunks, each of which contains spectra
+		 // corresponding to the same (scan-win, charge) pair. Within each chunk, the spectra should be sort by the MS2 scan.
+		 // The motivation here is to build per chunk (i.e. scan-win) map to extract chromatogram for precursor-fragment coelution.
+
+		 vector<SpectrumCollection::SpecCharge> spec_charge_chunk;
+		 int curr_precursor_mz = 0;
+
+		 for (vector<SpectrumCollection::SpecCharge>::const_iterator sc_chunk = spec_charges->begin();sc_chunk < spec_charges->begin() + (spec_charges->size()); sc_chunk++) {
+			++(*sc_index);
+			if (print_interval > 0 && *sc_index > 0 && *sc_index % print_interval == 0) { carp(CARP_INFO, "%d spectrum-charge combinations searched, %.0f%% complete", *sc_index, *sc_index / sc_total * 100); }
+
+			Spectrum* spectrum_chunk = sc_chunk->spectrum;
+			int precursor_mz_chunk = int(spectrum_chunk->PrecursorMZ());
+
+			// deal with a chunk if it's either the end of the same mz or it's the last element
+			if (((precursor_mz_chunk != curr_precursor_mz) || (sc_chunk == (spec_charges->begin() + spec_charges->size()-1)))
+					&& (spec_charge_chunk.size() > 0) ) {
+				// carp(CARP_DETAILED_DEBUG, "curr_precursor_mz=%d\tspec_charge_chunk size=%d", curr_precursor_mz, spec_charge_chunk.size());
+
+
+				// changing from buildSpectraIndexFromIsoWindowOld to buildSpectraIndexFromIsoWindowNew step-by-step
```

Comment:
seems like you should remove all reference to buildSpectraIndexFromIsoWindowOld and rename buildSpectraIndexFromIsoWindowNew as buildSpectraIndexFromIsoWindow